src/java/us/kbase/templates/perl_client.vm.properties

```diff
@@ -115,67 +115,84 @@ sub new
-	} elsif (exists $arg_hash2{"user_id"}) {
-	    my $token = Bio::KBase::AuthToken->new(@args);
-	    if (!$token->error_message) {
-	        $self->{token} = $token->token;
-	    }
-	}
-	
-	if (exists $self->{token})
-	{
-	    $self->{client}->{token} = $self->{token};
-	}
+        my %arg_hash2 = @args;
+        if ( exists $arg_hash2{ token } ) {
+            $self->{ token } = $arg_hash2{ token };
+        }
+        elsif ( exists $arg_hash2{ user_id } ) {
+            my $token = Bio::KBase::AuthToken->new( @args );
+            if ( !$token->error_message ) {
+                $self->{ token } = $token->token;
+            }
+        }
+
+        if ( exists $self->{ token } ) {
+            $self->{ client }{ token } = $self->{ token };
+        }
     }
 #end
 
-    my $ua = $self->{client}->ua;	 
-    my $timeout = $ENV${empty_escaper}{CDMI_TIMEOUT} || (30 * 60);	 
-    $ua->timeout($timeout);
+    my $ua      = $self->{ client }->ua;
+    my $timeout = $ENV${empty_escaper}{ CDMI_TIMEOUT } || ( 30 * 60 );
+    $ua->timeout( $timeout );
     bless $self, $class;
-    #    $self->_validate_version();
     return $self;
 }
 
 #foreach( $module in $modules )
 #if( $any_async || $async_version)
-sub _check_job {
-    my($self, @args) = @_;
 # Authentication: ${method.authentication}
-    if ((my $n = @args) != 1) {
-        Bio::KBase::Exceptions::ArgumentValidationError->throw(error =>
-                                   "Invalid argument count for function _check_job (received $n, expecting 1)");
+sub _check_job {
+    my ( $self, @args ) = @_;
+
+    if ( ( my $n = @args ) != 1 ) {
+        Bio::KBase::Exceptions::ArgumentValidationError->throw(
+            error =>  "Invalid argument count for function _check_job "
+                      . "(received $n, expecting 1)"
+        );
     }
-    {
-        my($job_id) = @args;
-        my @_bad_arguments;
-        (!ref($job_id)) or push(@_bad_arguments, "Invalid type for argument 0 \"job_id\" (it should be a string)");
-        if (@_bad_arguments) {
-            my $msg = "Invalid arguments passed to _check_job:\n" . join("", map { "\t$_\n" } @_bad_arguments);
-            Bio::KBase::Exceptions::ArgumentValidationError->throw(error => $msg,
-                                   method_name => '_check_job');
-        }
-    }
-    my $result = $self->{client}->call($self->{url}, $self->{headers}, {
-        method => "${module.module_name}._check_job",
-        params => \@args});
-    if ($result) {
-        if ($result->is_error) {
-            Bio::KBase::Exceptions::JSONRPC->throw(error => $result->error_message,
-                           code => $result->content->{error}->{code},
-                           method_name => '_check_job',
-                           data => $result->content->{error}->{error} # JSON::RPC::ReturnObject only supports JSONRPC 1.1 or 1.O
-                          );
+
+    my ( $job_id ) = @args;
+
+    my @bad_arguments;
```

Comment:
`@_bad_arguments` doesn't need that underscore, especially since it gets reported to the user and is therefore not private!

Reply:
You can't access `@bad_arguments` outside `_check_job`, no.